packages/rocketchat-livechat/server/models/Users.js

```diff
@@ -67,6 +67,45 @@ RocketChat.models.Users.findOnlineUserFromList = function(userList) {
  * Get next user agent in order
  * @return {object} User from db
  */
+RocketChat.models.Users.getBotAgent = function() {
+	const query = {
+		status: {
+			$exists: true,
+			$ne: 'offline'
+		},
+		statusLivechat: 'available',
+		roles: ['bot', 'livechat-agent']
```

Comment:
on this query do you want the user to have bot `bot` and `livechat-agent` roles? if so it's not correcty, you should do `roles: { $all: ['bot', 'livechat-agent'] }` instead.